ctgan/synthesizers/base.py

```diff
@@ -57,13 +57,60 @@ def wrapper(self, *args, **kwargs):
 
 
 class BaseSynthesizer:
-    """Base class for all default synthesizers of ``CTGAN``.
-
-    This should contain the save/load methods.
-    """
+    """Base class for all default synthesizers of ``CTGAN``."""
 
     random_states = None
 
+    def __getstate__(self):
+        """Improve pickling state for ``BaseSynthesizer``.
+
+        Convert to ``cpu`` device before starting the pickling process in order to be able to
+        load the model even when used from an external tool such as ``SDV``. Also, if
+        ``random_states`` are set, store their states as dictionaries rather than generators.
+
+        Returns:
+            dict:
+                Python dict representing the object.
+        """
+        device_backup = self._device
+        self.set_device(torch.device('cpu'))
```

Comment:
Is it possible that this can have the undesired effect of converting the model to use a `CPU` when the user still wants to use a `GPU`. For example, say they save before training

Reply:
After we get the `state` and make a copy of it, we set it back to `gpu` or the device that was set before. (line 78)